[CS2113T-T09-2] BinBash

PureUsagi opened the pull request:
BinBash helps small retail business maintain a comprehensive list of their inventory. It is optimized for CLI users so that frequent changes to the inventory can be done faster by typing in commands.

64-1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,516 @@
+
+### Command Component
+
+![CommandClassDiagram](images/CommandClassDiagram.png)
```
Maybe can enlarge the image as the content is not really readable.

64-1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,516 @@
 # Developer Guide
```
The guide covers a wide range of topics essential for developers, from setting up the project environment to detailed descriptions of its components, features, and logging practices. This thorough approach is excellent!

64-1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,516 @@
+
+### Storage Component
+
+![Storage class diagram](images/StorageClassDiagram.png)
```
Maybe not necessary to add a diagram for just two classes.

awesomesjh replied:
Perhaps make use of `skinparam classAttributeIconSize 0` to convert the access modifier icons to characters
![Screenshot 2024-04-04 133910](https://github.com/nus-cs2113-AY2324S2/tp/assets/89649663/5754a5ea-aa05-4dce-876e-fe5839156df9)

isaaceng7 replied:
Consider using characters instead of icons for the access modifiers.
![Screenshot 2024-04-04 152440](https://github.com/nus-cs2113-AY2324S2/tp/assets/110762696/17396bf5-2c17-42a9-aace-35b5f04b5145)

isaaceng7 replied:
Storage class diagram is very comprehensive! Well done!!

64-1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,516 @@
+
+Logging plays a crucial role in providing insights into the runtime behavior of the application and diagnosing issues. In this project, we utilize the `java.util.logging` package for comprehensive logging functionality.
+
+### BinBashLogger Class
```
While logging is mentioned, maybe a dedicated section on common errors users might encounter, and troubleshooting steps could improve user experience, especially for new users unfamiliar with CLI applications.

yyangdaa commented:
The inclusion of diagrams, such as class diagrams and sequence diagrams, are crucial for conveying the underlying design and interaction patterns within the application, making it easier for new developers to grasp complex concepts and workflows.

yyangdaa commented:
In the Developer Guide, it would be good to include the boundary conditions of the parameters (i.e. Update the quantity of an Item -s need to be non-negative)

64-1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,516 @@
 
-## Product scope
-### Target user profile
+![DataComponent](images/DataComponent.png)
```
Maybe not necessary to add all the methods in the diagram?

yyangdaa commented:
Overall, the developer guide is comprehensive and helpful.

awesomesjh commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,516 @@
+
+The **Sequence Diagram** below shows how the components interact with each other for the scenario where the user issues the command `list`.
+
+![Sequence Diagram](images/OverallSequenceDiagram.png)
```
In OverallSequenceDiagram.puml, should `[isUserActive]` be `userActive` instead? 
![Screenshot 2024-04-04 134122](https://github.com/nus-cs2113-AY2324S2/tp/assets/89649663/635b187e-cfd9-4df7-8bcb-e36e8bfb0fa9)

awesomesjh commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,516 @@
+
+Below shows the sequence diagram of a `Parser` parsing user input, to return the corresponding `Command`.
+
+![ParseSequenceDiagram](images/ParseSequenceDiagram.png)
```
Consider using `hide footbox` to hide this portion
![Screenshot 2024-04-04 141337](https://github.com/nus-cs2113-AY2324S2/tp/assets/89649663/4c9d757c-bdd7-4cf5-b710-7a19e09d1346)

awesomesjh commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,516 @@
+![SearchAssistantSequenceDiagram](images/SearchAssistantSequenceDiagram.png)
+The search command calls on methods in the SearchAssistant class to perform queries.
+
+![SearchAssistantReferenceFrameDiagram](images/SearchAssistantReferenceFrameDiagram.png)
+Each method searches through a corresponding field.
```
```suggestion
![SearchAssistantReferenceFrameDiagram](images/SearchAssistantReferenceFrameDiagram.png)

Each method searches through a corresponding field.
```
Perhaps add a space between these 2 lines
![Screenshot 2024-04-04 135253](https://github.com/nus-cs2113-AY2324S2/tp/assets/89649663/4ec5d0ed-2bdb-4db8-8659-36d53c35d73a)

awesomesjh reviewed (commented):
Great job overall, your developer guide was clear and easy to read. Only some minor issues with the UML diagrams.

isaaceng7 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,516 @@
+### Search command
+
+![SearchAssistantSequenceDiagram](images/SearchAssistantSequenceDiagram.png)
+The search command calls on methods in the SearchAssistant class to perform queries.
```
Indentation issue.
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110762696/f06f7ef1-9a18-4a62-be48-8097e1b5da26)

isaaceng7 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,516 @@
+
+### Add Item to Inventory
+
+![AddSequenceDiagram](images/AddSequenceDiagram.png)
```
Complete and neat sequence diagram for addSequence.

isaaceng7 reviewed (commented):
Very well done DG!

nur-haziq committed:
Rectify checkstyle violation

nur-haziq committed:
AddCommand.java: Remove unused RegEx format

nur-haziq committed:
SearchAssistant.java: Add a case sensitive, equals search

nur-haziq committed:
Add check to prevent items from being added if the name already exists

nur-haziq committed:
AddCommandTest.java: Add test case to check if item name already exists

nur-haziq committed:
Update JavaDocs of findItemByName() method

nur-haziq committed:
Fix use of SearchAssistant

nur-haziq committed:
Update test case output

nur-haziq committed:
ItemList.java: Remove unused import

nur-haziq committed:
Update test case

nur-haziq committed:
Ignore `binbash>` in runtest.sh

nur-haziq committed:
Revert to old runtest.sh

nur-haziq committed:
Modify file comparison

nur-haziq committed:
Remove any lines starting with binbash> before comparing the files.

nur-haziq committed:
Revert to original runtest.sh

nur-haziq committed:
Remove `binbash>` in Ui's readline() method

nur-haziq committed:
Add `binbash> ` back to Ui and EXPECTED.txt

nur-haziq committed:
Modify test script to include duplicate items

nkotaa committed:
Declare AllCommandOptionDescriptions

nkotaa committed:
Allow completer to get option descriptions

nkotaa committed:
Refactor AddCommandParser

nkotaa committed:
Refactor RestockCommandParser

nkotaa committed:
Refactor SellCommandParser

nkotaa committed:
Refactor UpdateCommandParser

nkotaa committed:
Refactor SearchCommandParser

nkotaa committed:
Refactor ListCommandParser

nkotaa committed:
Refactor DeleteCommandParser

nkotaa committed:
Decouple Ui and Parser

nkotaa committed:
Remove unused constructor

nkotaa committed:
Fix negative results bug

nkotaa committed:
Update Bye section

YHWong20 committed:
Merge pull request #265 from imanamirshah/U-PPP

Added PPP

nkotaa committed:
Parser: Add Javadoc

nkotaa committed:
Change allCommandsOptionDescriptions to HashMap

nkotaa committed:
DG: Delete Outdated UISection

nkotaa committed:
Fix inaccurate error message

nkotaa committed:
Ui: Update Javadoc

nur-haziq committed:
Merge branch 'master' into BugFixes

nur-haziq committed:
Merge pull request #266 from nur-haziq/BugFixes

Prevent user from adding items with the same name

nkotaa committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-T09-2/tp into refactor-command-optdesc

PureUsagi committed:
Fix typo in ListCommandTest

PureUsagi committed:
Add tests for list methods in ItemListTest

PureUsagi committed:
Refactor to follow coding standard

PureUsagi committed:
Add test for toString and delete item twice

PureUsagi committed:
Add test for sort by profit

PureUsagi committed:
Refactor string formatting to follow coding standard

PureUsagi committed:
Add test for bye command

PureUsagi committed:
Add hasToSave assertions to delete command tests

PureUsagi committed:
Fix typo

PureUsagi committed:
Add tests for delete command parser

PureUsagi committed:
Resolve checkstyle issues

YHWong20 committed:
Merge pull request #268 from PureUsagi/TestCoverage

Add additional test for test coverage

nkotaa committed:
Update Ui class diagram

nkotaa committed:
UiClassDiagram: reflect jline library

nkotaa committed:
ui-linereader-enhancement.png: Delete file

nur-haziq committed:
UserGuide.md: Add section to define what an Item is

nur-haziq committed:
Merge branch 'master' into UpdateDocumentation

nur-haziq committed:
Upload Command Format image for UG

nur-haziq committed:
UserGuide.md: Add key definitions section for UG

nur-haziq committed:
Fix typo

nur-haziq committed:
Update UG

PureUsagi committed:
Add image for cmd window

PureUsagi committed:
Add image for run window

PureUsagi committed:
Update windows subsection for the getting started section in UG

PureUsagi committed:
Update getting started section of UG

nkotaa committed:
DG: Update Ui Component Section

nkotaa committed:
Update UiClassDiagram

nkotaa committed:
Merge pull request #267 from nkotaa/refactor-command-optdesc

Refactoring and Bug Fixes

PureUsagi committed:
Rename image names of cmd and run

PureUsagi committed:
Add image for entering cd in windows cmd

PureUsagi committed:
Add folder navigation section for windows in UG with image

YHWong20 committed:
Merge pull request #270 from PureUsagi/UpdateUG-GettingStartedWindows

Add Images for Getting Started - Windows section of UG

YHWong20 committed:
Add CSS formatting for callout boxes

nkotaa committed:
Ui: Improve Defensiveness

YHWong20 committed:
Update command style formats for optional flags/args

YHWong20 committed:
Add sample images for mac - getting started

YHWong20 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-T09-2/tp into feature-finalUpdates

nkotaa committed:
Fix empty string invalid command bug

nkotaa committed:
Ui: Distinguish EOF & Interrupt exceptions

nur-haziq committed:
UserGuide.md: Replace 'integer' with 'whole number' for clarity

nur-haziq committed:
Merge pull request #269 from nur-haziq/UpdateDocumentation

Add 'Key Definitions' section to the UG

nkotaa committed:
Distinguish EOF & Interrupt handling

nkotaa committed:
Change format of warning messages

nkotaa committed:
Ui: Update Javadoc

nkotaa committed:
CommandOptionAdder: use assertions

nkotaa committed:
SearchCommandParser: remove hasOption variable

YHWong20 committed:
Merge and update formatting

YHWong20 committed:
Merge pull request #272 from YHWong20/feature-finalUpdates

Update UG formatting

nkotaa committed:
SearchCommandParser: extract parsePriceRange method

YHWong20 committed:
Replace md table with HTML

YHWong20 committed:
Merge pull request #273 from YHWong20/feature-finalUpdates

Replace md table with HTML

nkotaa committed:
SearchCommandParser: extract parseQuantityRange method

nkotaa committed:
SearchCommandParser: extract parseDateRange method

nkotaa committed:
Add unit tests to SearchCommandParser

nkotaa committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-T09-2/tp into refactor-command-optdesc

imanamirshah committed:
Update README.md

imanamirshah committed:
Update README.md

PureUsagi committed:
Refactor ByeCommand Test to use test setup.

PureUsagi committed:
Refactor DeleteCommandTest to use test setup

PureUsagi committed:
Refactor ListCommandTest to use test setup and reduce method dependancy

nkotaa committed:
Parser: resolve inconsistencies

PureUsagi committed:
Refactor ItemListTest to use test set up

imanamirshah committed:
Modify log message on console

imanamirshah committed:
Modify console log text

PureUsagi committed:
Refactor ItemComparatorByCostPriceTest by adding test setup and test class variable

PureUsagi committed:
Refactor ItemComparatorByExpiryDateTest by adding test setup and class variables

PureUsagi committed:
Refactor ItemComparatorBySalePriceTest by adding test setup and class variables

PureUsagi committed:
Refactor ItemComparatorByProfitTest by adding test setup

nkotaa committed:
Parser: add checkDuplicateOption method

PureUsagi committed:
Refactor ItemListTest to use test setup

PureUsagi committed:
Refactor DeleteCommandParserTest to use test setup

PureUsagi committed:
Refactor ListCommandParserTest to use test setup

nkotaa committed:
Update noOption test

nkotaa committed:
Update parsers to reject duplicate options

Our application allows for specifying duplicate options, silently taking
the first argument specified.

This is error prone and might lead to uncaught incorrect user inputs.

Lets be more explicit in warning users about duplicate options so that
they can specify actions more accurately.

PureUsagi committed:
Resolve typo and add test for full test coverage

imanamirshah committed:
Add Ui.png to README

imanamirshah committed:
Update README.md

nkotaa committed:
ParserTest: add unit tests

PureUsagi committed:
Merge pull request #276 from imanamirshah/U-README

Update README.md

nkotaa committed:
DG: Update UI section

imanamirshah committed:
Update PPP

- Edited portfolio hyperlinks for all members on AboutUs.md

imanamirshah committed:
Merge pull request #277 from imanamirshah/fix-Storage

Modify log message on console

nkotaa committed:
Update ParserClassDiagram

nur-haziq committed:
Merge pull request #278 from PureUsagi/Refactoring-Test&ItemList

Refactor Tests to use test set up and class variables

nur-haziq committed:
Merge pull request #279 from imanamirshah/U-PPP

Update PPP

nkotaa committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-T09-2/tp into refactor-command-optdesc

imanamirshah committed:
Modify PPP

nkotaa committed:
UG: update Linux getting started

nkotaa committed:
DG: Update acknowledgements

imanamirshah committed:
Update AboutUs.md

nkotaa committed:
DG: Add search manual test section

nkotaa committed:
DG: Add command completion manual test section

nur-haziq committed:
Merge pull request #280 from imanamirshah/U-PPP

Modify PPP

nur-haziq committed:
Merge branch 'master' into UpdateDocumentation

XavierLiau committed:
Updated QuoteCommand

nur-haziq committed:
Reorder layout of classes in Command Class Diagram

nur-haziq committed:
Upload new CommandClassDiagram.png image

XavierLiau committed:
Merge branch 'master' of https://github.com/XavierLiau34/tp

XavierLiau committed:
Fixed Expected.txt error

nur-haziq committed:
Update links present in nur-haziq.md

nur-haziq committed:
Rephrase link after 'Getting Started' section

XavierLiau committed:
Updated Quote Command

XavierLiau committed:
Added Tests for Search Command

XavierLiau committed:
Increased Test Coverage

XavierLiau34 committed:
Merge branch 'master' into master

XavierLiau committed:
Fixed Test Coverage

XavierLiau committed:
Merge branch 'master' of https://github.com/XavierLiau34/tp

nkotaa committed:
DG: Update ParserClassDiagram explanation

nkotaa committed:
DG: Remove un-developer like explanations

nkotaa committed:
SearchCommandParser: remove range index literals

nkotaa committed:
Update SearchCommandParser tests

nkotaa committed:
SearchCommandParser: improve test coverage

nkotaa committed:
Resolve checkstyle

nkotaa committed:
Move set user inactive to Bye execute

nkotaa committed:
Merge pull request #275 from nkotaa/refactor-command-optdesc

Improve Program Defensiveness

PureUsagi committed:
Rename windows images

PureUsagi committed:
Update UG to reflect image name change

PureUsagi committed:
Update tip to include restock and sell command in description

PureUsagi committed:
Add warning to list command to let user know to use latest list

PureUsagi committed:
Update warining in list command intro

PureUsagi committed:
Add warning for list index in sell command section

PureUsagi committed:
Add warning for list index in restock command section

PureUsagi committed:
Add warning for list index in update command section

PureUsagi committed:
Add waring for list index in delete command section

PureUsagi committed:
Add community engagement section to PPP

PureUsagi committed:
Add summary of contributions line to PPP

PureUsagi committed:
Update command feature description in PPP

PureUsagi committed:
Fix grammer issue

PureUsagi committed:
Add additional contributions to PPP

PureUsagi committed:
Remove UG and DG extracts to follow team standard

PureUsagi committed:
Merge pull request #283 from nur-haziq/UpdateDocumentation

Minor changes to documentation

PureUsagi committed:
Merge pull request #284 from PureUsagi/FinaliseDocs-UG&PPP

Add list index warnings to UG and Update PPP.

XavierLiau34 committed:
Merge branch 'master' into master

XavierLiau34 committed:
Update Parser.java

XavierLiau committed:
Updated Test Coverage

XavierLiau committed:
Fixed editing errors

nkotaa committed:
nkotaa.md: Update PPP

nkotaa committed:
nkotaa.md: fix formatting

nur-haziq committed:
Merge branch 'master' into UpdateUG

nur-haziq committed:
UserGuide.md: Add page breaks to improve readability

XavierLiau committed:
Increased Test Coverage

YHWong20 committed:
Merge pull request #257 from XavierLiau34/master

Added Quote Command

YHWong20 committed:
Merge pull request #286 from nkotaa/refactor-command-optdesc

nkotaa.md PPP

YHWong20 committed:
Merge pull request #287 from nur-haziq/UpdateUG

Minor changes to User Guide

YHWong20 committed:
Update callout box formatting

XavierLiau34 committed:
Rename XavierLiau34.md to xavierliau34.md

XavierLiau34 committed:
Update AboutUs.md

nur-haziq committed:
Merge branch 'master' into UpdateUG

YHWong20 committed:
Update table of contents

imanamirshah committed:
Update UserGuide.md

YHWong20 committed:
Merge pull request #288 from imanamirshah/U-UserGuide

Update UserGuide.md

YHWong20 committed:
Merge Iman's changes

YHWong20 committed:
Update descriptions and fix formatting

XavierLiau committed:
Increased Test Coverage

XavierLiau34 committed:
Merge pull request #289 from YHWong20/feature-finalUpdates

Update UG descriptions, fix formatting

YHWong20 committed:
Fix HTML rendering issue

YHWong20 committed:
Fix HTML rendering issue

YHWong20 committed:
Merge pull request #291 from YHWong20/feature-finalUpdates

Feature final updates

YHWong20 committed:
Add warning callout box in update section

YHWong20 committed:
Merge pull request #292 from YHWong20/feature-finalUpdates

Add warning callout box in update section

YHWong20 committed:
Update Parser component in DG

nur-haziq committed:
Merge branch 'master' into UpdateUG

XavierLiau34 committed:
Update Quotes.java

nur-haziq committed:
UserGuide.md: Update glossary section with relevant hyperlinks

XavierLiau34 committed:
Merge pull request #290 from XavierLiau34/master

Increased Test Coverage

nur-haziq committed:
Minor updates to PPP

nur-haziq committed:
Merge branch 'master' into BugFixes

nur-haziq committed:
Add ArrayIndexOutOfBounds exception catching

nur-haziq committed:
Storage.java: Add comma in JavaDocs for clarity

YHWong20 committed:
Update docs/diagrams/ParserLibraryClassDiagram.puml

Co-authored-by: nkotaa <110744303+nkotaa@users.noreply.github.com>

YHWong20 committed:
Update docs/diagrams/ParserLibraryClassDiagram.puml

Co-authored-by: nkotaa <110744303+nkotaa@users.noreply.github.com>

YHWong20 committed:
Update docs/diagrams/ParserLibraryClassDiagram.puml

Co-authored-by: nkotaa <110744303+nkotaa@users.noreply.github.com>

YHWong20 committed:
Update ParserLibraryClassDiagram

YHWong20 committed:
Merge commit 'refs/pull/293/head' of https://github.com/AY2324S2-CS2113T-T09-2/tp into feature-finalUpdates

YHWong20 committed:
Merge pull request #294 from nur-haziq/UpdateUG

Update Glossary section of UG

YHWong20 committed:
Merge pull request #293 from YHWong20/feature-finalUpdates

Update Parser component in DG

nur-haziq committed:
Merge branch 'master' into BugFixes

nur-haziq committed:
ItemList.java: Add JavaDocs and reduce horizontal length of methods

nur-haziq committed:
Storage.java: Remove unnecessary comments

nur-haziq committed:
RestockCommand.java: Remove unused RegEx format

nur-haziq committed:
SellCommand.java: Remove unused RegEx format

nur-haziq committed:
Parser.java: Add explicit 'Fallthrough' statement for certain cases

nur-haziq committed:
UpdateCommandParser.java: Rename method to sound more 'verb'-like

nur-haziq committed:
CommandCompleter.java: Remove blank lines

nur-haziq committed:
Remove unused imports

YHWong20 committed:
Resize Images

nur-haziq committed:
Merge pull request #296 from YHWong20/feature-finalUpdates

Resize UG Images

nur-haziq committed:
Merge branch 'master' into UpdateUG

YHWong20 committed:
Add storage format image

YHWong20 committed:
Merge pull request #297 from YHWong20/feature-finalUpdates

Add storage format image

PureUsagi committed:
Merge pull request #295 from nur-haziq/BugFixes

Minor changes to Code Quality

nur-haziq committed:
UserGuide.md: Add timely spacing to UG

YHWong20 committed:
Create _config.yml

PureUsagi committed:
Merge pull request #298 from nur-haziq/UpdateUG

Add pagebreak and spaces to UG

nur-haziq committed:
Merge branch 'master' into AddPPP

YHWong20 committed:
Update PPP for Yi Hao

YHWong20 committed:
Merge pull request #299 from YHWong20/feature-finalUpdates

Update PPP for Yi Hao

nur-haziq committed:
nur-haziq.md: Update PPP to shorten content

PureUsagi committed:
Merge pull request #300 from nur-haziq/AddPPP

Shorten PPP

nur-haziq committed:
nur-haziq.md: Update PPP

nur-haziq committed:
Remove unnecessary sections

nur-haziq committed:
Merge pull request #301 from nur-haziq/AddPPP

Remove unnecessary sections in PPP

imanamirshah committed:
Modify output message for `updateItemByIndex()`

nur-haziq committed:
Merge branch 'master' into UpdateUG

nur-haziq committed:
Capitalize placeholders to be consistent with command format

nur-haziq committed:
Merge pull request #302 from imanamirshah/fix-Update

Modify output message for `updateItemByIndex()`

nur-haziq committed:
Merge pull request #303 from nur-haziq/UpdateUG

Capitalize placeholders in UG

YHWong20 committed:
Add terminal-based examples for list sorted

YHWong20 committed:
Fix item name

nur-haziq committed:
Merge pull request #304 from YHWong20/feature-finalUpdates

Add terminal-based examples for list sorted
